code/game/gamemodes/gang/gang_items.dm

```diff
@@ -348,19 +692,142 @@
 /datum/gang_item/equipment/wetwork_boots
 	name = "Wetwork boots"
 	id = "wetwork"
-	cost = 20
+	cost = 15
 	item_path = /obj/item/clothing/shoes/combat/gang
 
+//////// REVIVIFICATION SERUM //////////
+
+/datum/gang_item/equipment/reviver
+	name = "Outlawed Reviver Serum"
+	id = "reviver"
+	cost = 50
+	item_path = /obj/item/weapon/reviver
+
+/datum/gang_item/equipment/reviver/get_cost(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	cost = 5
+	for(var/datum/mind/M in gang.gangsters)
+		if(M.current.stat != DEAD)
+			cost += 2
+	return cost
+
+/obj/item/weapon/reviver
+	name = "outlawed revivification serum"
+	desc = "Banned due to side effects of extreme rage, reduced intelligence, and violence. For gangs, that's just a fringe benefit."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "implanter1"
+	item_state = "syringe_0"
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_SMALL
+	origin_tech = "materials=2;biotech=5"
+	materials = list(MAT_METAL=600, MAT_GLASS=200)
+
+
+/obj/item/weapon/reviver/attack(mob/living/carbon/human/H, mob/user)
+	if(!ishuman(H) || icon_state == "implanter0")
+		return ..()
+	user.visible_message("<span class='warning'>[user] begins inject [H] with [src].</span>", "<span class='warning'>You begin to inject [H] with [src]...</span>")
+	var/total_burn	= 0
+	var/total_brute	= 0
+	H.notify_ghost_cloning("You're being injected with a revivification serum - return to your body!")
+	if(do_after(user, 80, target = H))
+		if(H.stat == DEAD && icon_state == "implanter1")
+			H.visible_message("<span class='warning'>[H]'s body thrashes violently.")
+			playsound(src, "bodyfall", 50, 1)
+			H.spin(20, 1)
+			if (H.suiciding || (H.disabilities & NOCLONE) || !H.getorgan(/obj/item/organ/heart) || !H.getorgan(/obj/item/organ/brain) || !H.mind)
+				H.visible_message("<span class='warning'>[H]'s body falls still again, they're gone for good.")
+				return
+			if(!H.key)
+				var/identity
+				var/mob/dead/observer/winner
+				H.visible_message("<span class='warning'>[H]'s body twitches as a spirit seeks to return to this broken form.")
+				if(H.mind.gang_datum)
+					identity = "[H.mind.gang_datum.name] gangster"
+				else
+					identity = "Vigilante"
+				var/list/mob/dead/observer/candidates = pollGhostCandidates("Would you to play the role of a revived [identity]?", "pAI", null, FALSE, 100)
+				if(LAZYLEN(candidates) && !QDELETED(src) && icon_state == "implanter1")
+					winner = pick(candidates)
+					H.key = winner.key
+				else
+					H.visible_message("<span class='warning'>[H]'s body falls still again, their spirit has moved on.")
+					return
+			total_brute = H.getBruteLoss()
+			total_burn = H.getFireLoss()
+			var/overall_damage = total_brute + total_burn + H.getToxLoss() + H.getOxyLoss()
```

Comment:
Its how defibs handled it, so that you get some of your original damage back (i.e. severe burns if you burnt to death)